packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1689,7 +2085,7 @@ export class TransactionController extends BaseControllerV1<
     filterToCurrentNetwork?: boolean;
     limit?: number;
   } = {}): TransactionMeta[] {
-    const chainId = this.getChainId();
+    const chainId = this.getChainId(); // TODO(JL): This should be made into an optional param
```

Comment:
should probably address this

Reply:
This is gross. Not sure how to work in an optional `chainId` param when `filterToCurrentNetwork` exists

Reply:
https://github.com/MetaMask/core/pull/3862 attempt